.github/workflows/rust.yml

```diff
@@ -90,28 +87,14 @@ jobs:
         TARGET: x86_64-unknown-linux-gnu
       run: sh ci/run.sh
 
-  # These jobs doesn't actually test anything, but they're only used to tell
-  # bors the build completed, as there is no practical way to detect when a
-  # workflow is successful listening to webhooks only.
-  #
-  # ALL THE PREVIOUS JOBS NEED TO BE ADDED TO THE `needs` SECTION OF THIS JOB!
-
-  end-success:
-    name: bors build finished
-    if: github.event.pusher.name == 'bors' && success()
-    runs-on: ubuntu-latest
-    needs: [miri, rustfmt_clippy, test, msrv]
-
-    steps:
-      - name: Mark the job as successful
-        run: exit 0
-
-  end-failure:
-    name: bors build finished
-    if: github.event.pusher.name == 'bors' && (failure() || cancelled())
+  conclusion:
+    needs: [test, msrv]
+    # !cancelled() executes the job regardless of whether the previous jobs passed or failed
+    if: ${{ !cancelled() }}
```

Comment:
Can you comment why we need this?
Maybe because it's not executed if previous jobs were skipped?

Reply:
I copied it from another repo that uses merge queues :D But yes, the issue is that if a job in `needs` would get skipped, `conclusion` would get skipped too, which we do not want. That is essentially what the comment already states, I think.

Reply:
I see. The comment doesn't mention skipped! :)

Reply:
Good point, expanded the comment.